src/server/array_func.c

```diff
@@ -420,18 +419,21 @@ chk_array_doneness(job *parent)
 		/*
 		 * Check if the history of the finished job can be saved or it needs to be purged .
 		 */
+		ptbl->tkm_flags |= TKMFLG_CHK_ARRAY;
+
 		svr_saveorpurge_finjobhist(parent);
 	} else {
 		/* Before we do a full save of parent, recalculate "JOB_ATR_array_indices_remaining" here*/
-		attribute *premain = &parent->ji_wattr[(int)JOB_ATR_array_indices_remaining];
-		if (premain->at_flags & ATR_VFLAG_MODCACHE) {
+		if (ptbl->tkm_flags & TKMFLG_REVAL_IND_REMAINING) {
+			attribute *premain = &parent->ji_wattr[(int)JOB_ATR_array_indices_remaining];
```

Comment:
The same code is being used in stat_job.c as well. Should this be a common function?

Reply:
addressed in commit named rc1  https://github.com/PBSPro/pbspro/pull/807/commits/b2f22afcadb50bc872cb198d65b51190511aaf45